### PR TITLE
oscclip: init at 0.4.1

### PR DESCRIPTION
**Pull request by traxys**
###### Description of changes

This add the https://github.com/rumpelsepp/oscclip package, that allows to use osc-52 control sequences to copy/paste in a display manager (Xorg/wayland) agnostic way

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by traxys** [comment]
I have no idea what is the error in this PR, as the package seems to build correctly

**Comment by GaetanLepage** [comment]
> I have no idea what is the error in this PR, as the package seems to build correctly

To give a little more context, the package builds succesfully:
```
nixpkgs on  oscclip
❯ nix-build . --check -A oscclip                                      
checking outputs of '/nix/store/lvypwgbarp2ynaib5li71q9kfv5lmxd4-python3.10-oscclip-0.4.1.drv'...
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing pip-build-hook
Using pipBuildPhase
Using pipShellHook
Sourcing pip-install-hook
Using pipInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
unpacking sources
unpacking source archive /nix/store/j85fcliafgsa4rnzc6wkgavyrz589hyj-source
source root is source
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file source/setup.cfg
patching sources
Removing path dependencies
Finished removing path dependencies
Removing git dependencies
Finished removing git dependencies
configuring
no configure script, doing nothing
building
Executing pipBuildPhase
Creating a wheel...
WARNING: The directory '/homeless-shelter/.cache/pip' or its parent directory is not owned or is not writable by the current user. The cache has been disabled. Check the permissions and owner of that directory. If executing pip with sudo, you should use sudo's -H flag.
Processing /build/source
  Running command Preparing metadata (pyproject.toml)
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: oscclip
  Running command Building wheel for oscclip (pyproject.toml)
  Building wheel for oscclip (pyproject.toml) ... done
  Created wheel for oscclip: filename=oscclip-0.4.1-py3-none-any.whl size=16157 sha256=78fbd81f1c2e93719b5f1765eb001edd69da64eb8155426763ac2ad86cc7c5f9
  Stored in directory: /build/pip-ephem-wheel-cache-5zowi0rz/wheels/79/4b/9b/fc24941d75fdb959a71ee80195b2c89d6ae9389f9f0f80fb22
Successfully built oscclip
Finished creating a wheel...
Finished executing pipBuildPhase
installing
Executing pipInstallPhase
/build/source/dist /build/source
Processing ./oscclip-0.4.1-py3-none-any.whl
Installing collected packages: oscclip
Successfully installed oscclip-0.4.1
/build/source
Finished executing pipInstallPhase
pythonOutputDistPhase
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/5zhxp5yw7j2zj2wwsnyc9554bmnsgisi-python3.10-oscclip-0.4.1
strip is /nix/store/wn31i3dzwahz6ccws8bs1nwyqrpgsvj7-gcc-wrapper-11.3.0/bin/strip
stripping (with command strip and flags -S) in  /nix/store/5zhxp5yw7j2zj2wwsnyc9554bmnsgisi-python3.10-oscclip-0.4.1/lib /nix/store/5zhxp5yw7j2zj2wwsnyc9554bmnsgisi-python3.10-oscclip-0.4.1/bin
patching script interpreter paths in /nix/store/5zhxp5yw7j2zj2wwsnyc9554bmnsgisi-python3.10-oscclip-0.4.1
checking for references to /build/ in /nix/store/5zhxp5yw7j2zj2wwsnyc9554bmnsgisi-python3.10-oscclip-0.4.1...
shrinking RPATHs of ELF executables and libraries in /nix/store/qvn10r93rnhq71hwzac98zkhxj3vssi8-python3.10-oscclip-0.4.1-dist
strip is /nix/store/wn31i3dzwahz6ccws8bs1nwyqrpgsvj7-gcc-wrapper-11.3.0/bin/strip
patching script interpreter paths in /nix/store/qvn10r93rnhq71hwzac98zkhxj3vssi8-python3.10-oscclip-0.4.1-dist
checking for references to /build/ in /nix/store/qvn10r93rnhq71hwzac98zkhxj3vssi8-python3.10-oscclip-0.4.1-dist...
Rewriting #!/nix/store/al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/bin/python3.10 to #!/nix/store/al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9
wrapping `/nix/store/5zhxp5yw7j2zj2wwsnyc9554bmnsgisi-python3.10-oscclip-0.4.1/bin/osc-copy'...
Rewriting #!/nix/store/al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9/bin/python3.10 to #!/nix/store/al6g1zbk8li6p8mcyp0h60d08jaahf8c-python3-3.10.9
wrapping `/nix/store/5zhxp5yw7j2zj2wwsnyc9554bmnsgisi-python3.10-oscclip-0.4.1/bin/osc-paste'...
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
running install tests
no Makefile or custom installCheckPhase, doing nothing
pythonCatchConflictsPhase
pythonRemoveBinBytecodePhase
pythonImportsCheckPhase
Executing pythonImportsCheckPhase
/nix/store/5zhxp5yw7j2zj2wwsnyc9554bmnsgisi-python3.10-oscclip-0.4.1
```

However, the `nixpkgs-review pr 202587` fails on some weird TOML error.
Full output: https://gist.github.com/GaetanLepage/a8dfa7b177cf0d7e3be7a51cb47b272f

**Comment by traxys** [comment]
Sorry failed a rebase....

File: pkgs/tools/misc/oscclip/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  format = "pyproject";
+
+  src = fetchFromGitHub {
+    owner = "rumpelsepp";
+    repo = "oscclip";
+    rev = "v${version}";
+    sha256 = "sha256-WQvZn+SWamEqEXPutIZVDZTIczybtHUG9QsN8XxUeg8=";
+  };
+
+  buildInputs = with python3Packages; [ poetry-core ];
```
Comment (treed) [inline pkgs/tools/misc/oscclip/default.nix:21]: As this is a build-time dependency, I think this should be `nativeBuildInputs`.

I wasn't sure, so I did a quick grep. I only found one case that isn't marked as a native input.

Comment (traxys) [reply pkgs/tools/misc/oscclip/default.nix]: Yeah it's a build dependency, fixed it.

Comment (treed) [reply pkgs/tools/misc/oscclip/default.nix]: Other than that, it LGTM, but I'm still pretty new myself.

File: pkgs/tools/misc/oscclip/default.nix [hunk 2]
```diff
@@ -0,0 +1,36 @@
+{ fetchFromGitHub
+, python3Packages
+, stdenv
+, writeText
+, lib
+, ...
+}:
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/oscclip/default.nix:7]:
```suggestion
}:

```

File: pkgs/tools/misc/oscclip/default.nix [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  meta = with lib; {
+    description = "A program that allows to copy/paste from a terminal using osc-52 control sequences.";
+    longDescription = ''
+      oscclip provides two commands: osc-copy and osc-paste. These commands allow to interact with the clipboard through the terminal directly.
+      This means that they work through ssh sessions for example (given that the terminal supports osc-52 sequences).
+    '';
+
+    homepage = "https://github.com/rumpelsepp/oscclip";
+
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/oscclip/default.nix:31]:
```suggestion
    homepage = "https://github.com/rumpelsepp/oscclip";
```

File: pkgs/tools/misc/oscclip/default.nix [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    license = licenses.gpl3Only;
+    maintainers = [ maintainers.traxys ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/oscclip/default.nix:34]:
```suggestion
```
buildPythonApplication defaults to linux+darwin which is only supported by python

**Comment by traxys** [comment]
Made the changes, hope I did not typo anything

**Pull request merged**